### DISCUSSION TITLE
Martensite Parent Grain Statistics

### DISCUSSION THREAD

**Discussion by fischertim1** [mtex-toolbox/mtex#1521] [Ask Anything]
Hello,

I'm just getting familiar with the new "Martensite Parent Grain Reconstruction" tool. So far it works quite well for my case. I was wondering, what is the best method to determine the PAG size, packet size and block width? 
I would appreciate some suggestions on this.

**Comment by nyyssont** [comment]
Hello,

unfortunately the choice of "correct" grain size metric depends quite heavily on the context. 

In industry, access to EBSD data is typically not as good and the data for parent grains most often comes from optical metallography of selectively etched surfaces. In such a case, the twin boundaries in parent austenite are not revealed. Moreover, the grain size is typically estimated as some version of mean linear intercept, or even by visual comparison to standard metallographs. In such cases, a meaningful comparison to old data would be by doing mean linear intercepts (manual or automated) on parent austenite grain boundaries, ignoring twin boundaries. Equivalent circle diameter (ECD) can also be used, since it often correlates nicely with mean linear intercept. Note that the main motivation of the choice here is to enable useful discussion of the results with people more used to optical metallography.

For more scientific work, it is important to try to figure out how the data you are comparing your results to has been obtained. Generally speaking, any PAG results before 2007 are quite likely to have been determined by selective etching and optical metallography. Same considerations as above apply in those cases.

For other considerations, I really have no good rule of thumb. ECD with both area-weighing and arithmetic mean are often used to see if correlations with other data can be found.

Block width is a special case, since it can be roughly related to the crystallography of the phase transformation. One description can be found here:
https://www.sciencedirect.com/science/article/pii/S0921509306006058

ORTools has a feature for looking at block width this way. If you are interested in writing a paper on the topic, it would be a good idea to describe your block width measurement algorithm in detail, since no standardized method exists for determining it (I at least would appreciate  it a lot).

**Reply by fischertim1** [reply]
Hello, 

thank you very much for the detailed answer, it helps me a lot.
I have another question regarding the 'Martensite Parent Grain Reconstruction Tool':
Is it also possible to plot the number of packets in each individual prior austenite grain as a function of the corresponding prior austenite grain size?

Many thanks

**Reply by nyyssont** [reply]
Hello,

I am quite sure what you ask is possible to do. I don't have any code to share, but the steps would be:

1. Reconstruct the austenite orientation map
2. Determine variant and packet id for each martensite domain
3. Merge the now austenitic grains into a parent austenite grain map
4. You should end up with a list with:
- original grain id
- parent grain id
- packet id

From this, it should be possible to count the number of packets in each parent grain.

**Reply by fischertim1** [reply]
Hej Tuomo,

thank you very much for your last response.
I was wondering, if you have any code to share for excluding twin boundaries in parent austenite?
I'm struggling with that a bit at the moment. Many thanks again.

**Comment by nyyssont** [comment]
Hello, attaching a minimal example here for everyone's benefit:

```matlab
mtexdata martensite

plotx2east

% grain reconstructions
[grains,ebsd('indexed').grainId] = calcGrains(ebsd('indexed'), 'angle', 3*degree,'unitCell')

ebsd(grains(grains.grainSize < 2)) = 'notIndexed'; % Remove single-pixel measurements
ebsd('Iron fcc') = 'notIndexed';     % remove redundant austenite
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'angle',3*degree,'qhull') % construct final grains object

%%
job = parentGrainReconstructor(ebsd,grains); % set up reconstructor
job.useBoundaryOrientations = 1; % use boundary orientations
job.p2c = orientation.GreningerTrojano(job.csParent, job.csChild); % set up initial OR
job.calcParent2Child('quantile',0.6,'c2c'); % determine representative OR
job.calcVariantGraph('threshold',5*degree,'tolerance',2*degree) % set up variant graph
job.clusterVariantGraph('numIter',5,'inflationPower',1) % cluster variant graph
job.calcParentFromVote % calculate parent orientations
%
job.calcGBVotes('p2c','reconsiderAll') % reconsider all orientations
job.calcParentFromVote % recalculate
job.mergeSimilar % merge grains with a similar parent orientation (5 degree threshold is default)

%% Determine twinning boundaries:

%List of austenite grain boundaries:
gB = job.grains.boundary('Iron fcc','Iron fcc');
%List of neighbors:
gP = unique(gB.grainId,'rows');

% List of grains with mutual twin boundary:
gP3 = gP(angle(inv(job.grains(gP(:,1)).meanOrientation).* ...
    job.grains(gP(:,2)).meanOrientation,CSL(3,job.p2c.CS)) < 5*degree,:);
% Corresponding actual boundaries:
gB3 = gB.selectByGrainId(gP3);

% Plot parent grains colored according orientation, twin gB cyan:
figure
plot(job.parentGrains,job.parentGrains.meanOrientation,'linewidth',1.5)
hold on
plot(gB3,'DisplayName','CSL 3','linecolor','cyan','linewidth',1.5)

%% Merge and plot grains without twins:

grainsMerged = merge(job.grains,gP3)

figure
plot(grainsMerged)
```
